packages/lit-auth-client/src/lib/providers/EthWalletProvider.ts

```diff
@@ -50,47 +54,92 @@ export default class EthWalletProvider extends BaseProvider {
   public async authenticate(
     options?: EthWalletAuthenticateOptions
   ): Promise<AuthMethod> {
-    const address = options?.address;
-    const signMessage = options?.signMessage;
-    const chain = options?.chain || 'ethereum';
+    return EthWalletProvider.authenticate({
+      signer: options,
+      address: options?.address,
+      chain: options?.chain,
+      litNodeClient: this.litNodeClient,
+      expiration: options?.expiration,
+      domain: this.domain,
+      origin: this.origin,
+    });
+  }
+
+  /**
+   * Generate a wallet signature to use as an auth method
+   * 
+   * @param {EthWalletAuthenticateOptions} options
+   * @param {string} [options.address] - Address to sign with
+   * @param {function} [options.signMessage] - Function to sign message with
+   * @param {string} [options.chain] - Name of chain to use for signature
+   * @param {number} [options.expiration] - When the auth signature expires
+   * @returns {Promise<AuthMethod>} - Auth method object containing the auth signature
+   * @static
+   * @memberof EthWalletProvider
+   * 
+   * @example
+   * ```typescript
+   *   const authMethod = await EthWalletProvider.authenticate({
+   *      signer: wallet,
+   *      litNodeClient: client,
+   *   });
+   * ```
+   */
+  public static async authenticate({
+    signer,
+    address,
+    chain,
+    litNodeClient,
+    expiration,
+    domain,
+    origin,
+  }: {
+    signer: ethers.Signer | ethers.Wallet | any;
```

Comment:
why do we union with `any` since by doing so we will essentially omit the typing of `ethers.Signer | ethers.Wallet`

Reply:
You are right - I've changed it to `EthWalletAuthenticateOptions` basically a eth signer implementation